ledgerblue/comm.py

```diff
@@ -209,6 +217,14 @@ def exchange(self, apdu, timeout=TIMEOUT):
 				possibleCause = "Unexpected state of device: verify that the right application is opened?"
 			if sw == 0x6e00:
 				possibleCause = "Unexpected state of device: verify that the right application is opened?"
+			if sw == 0x5515:
+				possibleCause = "Did you unlock the device?"
+			if sw == 0x6814:
+				possibleCause = "Unexpected target device: verify that your are using the right device?"
+			if sw == 0x511F:
+				possibleCause = "The OS version on your device does not seem compatible with the SDK version used to build the app"
+			if sw == 0x5120:
+				possibleCause = "Sideload is not supported on Nano X"
```

Comment:
Strange to duplicate the whole error codes check. Couldn't it be optimized in a common function?

Reply:
true --> I looked too quickly; I thought that some error codes differed depending on the communication method